Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -527,6 +530,106 @@ class Gx2Fitter {
 
           // We count the processed state
           ++result.processedStates;
+
+          // Update the number of holes count only when encountering a
+          // measurement
+          result.measurementHoles = result.missedActiveSurfaces.size();
+        } else if (surface->associatedDetectorElement() != nullptr ||
+                   surface->surfaceMaterial() != nullptr) {
+          // Here we handle material and holes
+          // TODO add material handling
+          ACTS_VERBOSE("Non-Measurement surface " << surface->geometryId()
+                                                  << " detected.");
+
+          // We only create track states here if there is already a measurement
+          // detected or if the surface has material (no holes before the first
+          // measurement)
+          if (result.measurementStates > 0
+              // || surface->surfaceMaterial() != nullptr
```

Comment:
that looks a bit funky 😄

Reply:
Yes, this is the logical place where we should handle the material soon